SimPEG/utils/model_utils.py

```diff
@@ -260,3 +260,69 @@ def ismember(a, b):
     actv[inds] = True
 
     return actv
+
+
+def depth_weighting(mesh, indActive=None, v=2, z0=None):
+    """
+    Create depth weighting
+
+    Parameters
+    ----------
+    mesh : Mesh object
+        discretize model space.
+    indActive : array of bool
+        index vector for the active cells on the mesh below the topography
+    v : float
+        exponent constant parameter for depth weighting.
+    z0 : scalar (float or int) or np.ndarray with shape of (points, dim)
+        adjustable constant parameter.
+
+    Returns
+    -------
+    wz : numpy.ndarray
+        depth weighting captures decay of the potential field data.
+
+
+    ..math::
+
+        w(z) = (z + z0) ** (-0.5*v)
+
+        where z is depth of model cells along the z direction;
+        v and z0 are two adjustable parameters.
+
+
+    Notes:
+
+        The depth weighting depends on the depth from receiver locations.
+
+
+      """
+
+
+    # Set up default value of z0
+    if z0 is None:
+        z0 = 0.5 * mesh.h_gridded.min()
```

Comment:
ok I think we are getting confused between @jcapriot and mine. So z0 will be the small constant to basically smooth the function and avoid zero division.
This is different than the topo/obs values (maybe reference_xyz?) that we to compute distances/depth from. It doesn't make much sense here to compute a distance from the smallest cell size...

Reply:
I see. So now we actually need three variables for this function: v (exponent value), reference_loc (scalar or 2d array to compute delta_z) and z0 (small constant avoiding zero in denominator). Does this make sense?

Reply:
sounds right, and only `mesh` and `reference_locs` are required, the `z0` and `exp` can be kwargs with default as you have it now.

Reply:
Sounds good. I will update right later. Thanks!

Reply:
sorry for being picky, but can we use plain English too for the variable names. Like `z0` -> `threshold`, `v` -> `exponent`

Reply:
No worries, sure, that does more explicit.